metrics/cli/test/flutter/adapter/flutter_cli_service_adapter_test.dart

```diff
@@ -55,14 +56,48 @@ void main() {
     test(
       ".build() enables web support before building the web application",
       () async {
-        await flutterCliServiceAdapter.build(path);
+        await flutterAdapter.build(path);
 
         verifyInOrder([
           flutterCli.enableWeb(),
           flutterCli.buildWeb(path),
         ]);
       },
     );
+
+    test(
+      ".version() rethrows the StateError if showing the version throws it",
+          () {
+        when(flutterCli.version()).thenThrow(stateError);
```

Comment:
Since the `version` method returns a `Future`, we should use the `thenAnswer` method to mock it.
```suggestion
        when(flutterCli.version()).thenAnswer((_) => Future.error(stateError));
```
Please, apply this suggestion to all tests.